benchmark/tools/compare.py

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env python3
+import sys
+import json
+import argparse
+import math
+import pandas as pd
+
+parser = argparse.ArgumentParser(description="Compare to Ginkgo benchmark outputs")
+parser.add_argument(
+    "--outliers", action="store_true", help="List outliers from the results"
+)
+parser.add_argument(
+    "--outlier-threshold",
+    type=float,
+    default=10,
+    help="At what percentage of deviation (above or below) should outliers be reported",
+)
+parser.add_argument(
+    "--outlier-count",
+    type=int,
+    default=1000,
+    help="How many outliers should be reported per benchmark",
+)
+parser.add_argument("--output", choices=["json", "csv", "markdown"], default="json")
+parser.add_argument("baseline")
+parser.add_argument("comparison")
+args = parser.parse_args()
+keys = {"stencil", "size", "filename", "n", "r", "k", "m"}
+comparison_keys = {"time", "storage", "iterations"}
+suffix = ".ratio"
+
+
+def key_to_json(key: tuple) -> dict:
+    """Restore a JSON output from a key tuple"""
+    result = {}
+    for key_name, key_val in zip(keys, key):
+        if key_val is not None:
+            result[key_name] = key_val
+    return result
+
+
+def key_to_str(key: tuple) -> str:
+    """Restore a JSON output string from a key tuple"""
+    mapped = key_to_json(key)
+    if "filename" in mapped.keys():
+        return mapped["filename"]
+    if "stencil" in mapped.keys():
+        return "stencil({}, {})".format(mapped["stencil"], mapped["size"])
```

Comment:
Which python version are we using. If python > 3.6 fstrings are more concise
```suggestion
        return f"stencil({mapped["stencil"]}, {mapped["size"]})"
```

Reply:
right, I'm still used to older python styles ;) I'll update those